packages/shared_preferences/shared_preferences_android/android/src/main/kotlin/io/flutter/plugins/sharedpreferences/StringListObjectInputStream.kt

```diff
@@ -0,0 +1,31 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.sharedpreferences
+
+import java.io.IOException
+import java.io.InputStream
+import java.io.ObjectInputStream
+import java.io.ObjectStreamClass
+
+/**
+ * An ObjectInputStream that only allows string lists, to prevent injected prefs from instantiating
+ * arbitrary objects.
+ */
+class StringListObjectInputStream(input: InputStream) : ObjectInputStream(input) {
+  @Throws(ClassNotFoundException::class, IOException::class)
+  override fun resolveClass(desc: ObjectStreamClass?): Class<*>? {
+    val allowList =
```

Comment:
We talked out of band and I am comfortable with this approach but I think this list is probably the most risky to non malicious apps. I wonder if we should apply this patch to G3 and see if any unit tests break as additional testing.